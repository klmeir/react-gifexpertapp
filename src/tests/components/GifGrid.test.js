import React from 'react';
import '@testing-library/jest-dom';

import { GifGrid } from '../../components/GifGrid';
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en el <GifGrid />', () => {

    const category = 'Bleach';

    test('debe mostrarse correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        
        const wrapper = shallow( <GifGrid category={category} /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('h3').text().trim() ).toBe( category );

    });
    
    test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
        
        const gifs = [
            {
                id: '123',
                title: 'Cualquier cosa',
                url: 'https://localhost/cualquier/cosa.jpg'
            },
            {
                id: '456',
                title: 'Otro elemento',
                url: 'https://localhost/otros/ping.jpg'
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow( <GifGrid category={category} /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe( false );
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });
    

});
