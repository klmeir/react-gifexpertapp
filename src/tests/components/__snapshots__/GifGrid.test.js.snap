// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Bleach
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="456"
      key="456"
      title="Otro elemento"
      url="https://localhost/otros/ping.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Bleach
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
